### PR TITLE
For krux in AMP: checking if local storage exists before using it

### PR DESCRIPTION
**Pull request by NataliaLKB**
AMP pages will soon be available within Newsstand. In the test our ads aren't loading. The only information I know for sure:
- Ads load in Guardian AMP pages for newsstand for IOS
- On android other third party embeds, like twitter, seem to work
- Some of the calls to Krux and DFP are working, just not the final ones

While talking to @maxspencer and @ChrisOwen101 I learnt that even though it is a webview, localstorage is not supported, and might not be handled by this app. Therefore I am adding a check to the amp remote file.
## Does this affect other platforms - Amp, Apps, etc?

Only affects AMP
## Request for comment

@rich-nguyen

### CODE REVIEW

File: facia/public/amp_remote.html [hunk 1]
```diff
@@ -74,7 +74,7 @@
         }
 
         function attachKruxSegments(config) {
-            if (localStorage.getItem('kxsegs') !== null) {
+            if (localStorage && localStorage.getItem('kxsegs') !== null) {
```
Comment (rich-nguyen) [inline facia/public/amp_remote.html]: [Here](https://github.com/guardian/frontend/pull/12983#discussion_r64717316) is a thread about local storage's wild behaviour.

Comment (rich-nguyen) [inline facia/public/amp_remote.html]: Yes, I have a made 2-chain pr reference.

**Comment by NataliaLKB** [comment]
@rich-nguyen if you get a chance, can you take another look? I have put in a more comprehensive check based on our `storage.js` file.

Even though there are potentially more things to fix to getting our ads working in web views, I think fixing things one at a time is best.

**Comment by rich-nguyen** [comment]
👍

**Pull request merged**